pkg/defkinds/php/extensions.go

```diff
@@ -383,55 +383,95 @@ func InstallExtensions(
-
 	cmds := []string{}
+	extensionNames := getExtensionNames(extensions)
+	runOpts := []llb.RunOption{
+		llb.WithCustomNamef("Install PHP extensions (%s)", strings.Join(extensionNames, ", "))}
+
+	coreExtensions := filterExtensions(extensions, isCoreExtension)
 	if len(coreExtensions) > 0 {
-		coreExtensionNames := getExtensionNames(coreExtensions)
-		coreExtensionSpecs := getCoreExtensionSpecs(coreExtensions)
+		cmds = append(cmds, installCoreExtensions(stageDef, coreExtensions)...)
+	}
 
-		cmds = append(cmds, configureExtBuilds(stageDef, coreExtensionNames)...)
-		cmds = append(cmds,
-			"docker-php-ext-install -j\"$(nproc)\" "+strings.Join(coreExtensionSpecs, " "))
+	peclExtensions := filterExtensions(extensions, isNotCoreExtension)
+	if len(peclExtensions) > 0 {
+		peclExtCmds, peclExtRunOpts := installPeclExtensions(
+			stageDef, peclExtensions, buildOpts)
 
-		if version.Compare(stageDef.MajMinVersion, "7.3", ">=") {
-			cmds = append(cmds, "docker-php-source delete")
-		}
+		cmds = append(cmds, peclExtCmds...)
+		runOpts = append(runOpts, peclExtRunOpts...)
 	}
-	if len(peclExtensions) > 0 {
-		peclExtensionNames := getExtensionNames(peclExtensions)
-		peclExtensionSpecs := getPeclExtensionSpecs(peclExtensions)
 
-		cmds = append(cmds,
-			"curl -f -o /usr/local/sbin/notpecl https://storage.googleapis.com/notpecl/notpecl",
-			"chmod +x /usr/local/sbin/notpecl")
+	runOpts = append(runOpts, llbutils.Shell(cmds...))
+	if buildOpts.IgnoreLayerCache {
+		runOpts = append(runOpts, llb.IgnoreCache)
+	}
 
-		isAlpine := stageDef.DefLocks.OSRelease.Name == "alpine"
-		if isAlpine {
-			cmds = append(cmds, "apk add --no-cache --virtual=.phpize $PHPIZE_DEPS")
-		}
+	return state.Run(runOpts...).Root()
+}
 
-		cmds = append(cmds,
-			"notpecl install "+strings.Join(peclExtensionSpecs, " "),
-			"docker-php-ext-enable "+strings.Join(peclExtensionNames, " "))
+func installCoreExtensions(
+	stageDef StageDefinition,
+	coreExtensions map[string]string,
+) []string {
+	coreExtensionNames := getExtensionNames(coreExtensions)
+	coreExtensionSpecs := getCoreExtensionSpecs(coreExtensions)
 
-		if isAlpine {
-			cmds = append(cmds, "apk del .phpize")
-		}
-		cmds = append(cmds, "rm -rf /usr/local/sbin/notpecl")
+	cmds := append(
+		configureExtBuilds(stageDef, coreExtensionNames),
+		"docker-php-ext-install -j\"$(nproc)\" "+strings.Join(coreExtensionSpecs, " "))
+
+	if version.Compare(stageDef.MajMinVersion, "7.3", ">=") {
+		cmds = append(cmds, "docker-php-source delete")
 	}
 
-	extensionNames := getExtensionNames(extensions)
-	runOpts := []llb.RunOption{
-		llbutils.Shell(cmds...),
-		llb.WithCustomNamef("Install PHP extensions (%s)", strings.Join(extensionNames, ", "))}
+	return cmds
+}
 
-	if buildOpts.IgnoreLayerCache {
-		runOpts = append(runOpts, llb.IgnoreCache)
+func installPeclExtensions(
+	stageDef StageDefinition,
+	peclExtensions map[string]string,
+	buildOpts builddef.BuildOpts,
+) ([]string, []llb.RunOption) {
+	cmds := []string{
+		// @TODO
```

Comment:
isn't it done ?

Reply:
Mh I should add a description to this TODO :stuck_out_tongue: Actually I put it to mark the URL to change when notpecl v0.1 will be properly released :wink: